src/librustc_data_structures/graph/dominators/mod.rs

```diff
@@ -118,6 +118,15 @@ impl<Node: Idx> Dominators<Node> {
         self.immediate_dominators[node].unwrap()
     }
 
+    /// Find the children of a node in the dominator tree.
+    pub fn immediately_dominates(&self, node: Node) -> Vec<Node> {
```

Comment:
I don't know if this will be possible (it probably depends on the borrow checker), but it would almost certainly be faster to make this an iterator rather than a producer of a `Vec`. You could probably get away with `-> impl Iterator<Node>` or similar.

Reply:
There was a reason why I didn't do this, but I don't recall what it was. Let me experiment and come back to you.